be/src/io/s3_random_access_file.cpp

```diff
@@ -0,0 +1,90 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
+
+#ifdef STARROCKS_WITH_AWS
+
+#include "io/s3_random_access_file.h"
+
+#include <aws/core/Aws.h>
+#include <aws/s3/S3Client.h>
+#include <aws/s3/model/GetObjectRequest.h>
+#include <aws/s3/model/HeadObjectRequest.h>
+#include <fmt/format.h>
+
+namespace starrocks::io {
+
+StatusOr<int64_t> S3RandomAccessFile::read(void* data, int64_t count) {
+    ASSIGN_OR_RETURN(auto nread, S3RandomAccessFile::read_at(_offset, data, count));
+    _offset += nread;
+    return nread;
+}
+
+StatusOr<int64_t> S3RandomAccessFile::read_at(int64_t offset, void* out, int64_t count) {
+    if (UNLIKELY(_size == -1)) {
+        ASSIGN_OR_RETURN(_size, S3RandomAccessFile::get_size());
+    }
+    if (offset < 0) {
+        return Status::InvalidArgument(fmt::format("Invalid offset {}", offset));
+    }
+    if (offset >= _size) {
+        return 0;
+    }
+    auto range = fmt::format("bytes={}-{}", offset, std::min<int64_t>(offset + count, _size));
+    Aws::S3::Model::GetObjectRequest request;
```

Comment:
why not move this call into `S3Client`？

Reply:
there is already a `S3Client` in StarRocks, why not use that?

Reply:
'IO' module was designed to be the most fundamental utility, so I tend to make it independent from others.
Make others depend on 'IO', not reversed.